packages/grid/x-grid-modules/src/components/error-boundary.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import { ErrorInfo } from 'react';
+import { Logger } from '../hooks/utils/useLogger';
+import { ApiRef } from '../models/api';
+
+export interface ErrorBoundaryProps {
+  logger: Logger;
+  render: ({ error }: any) => React.ReactNode;
+  api: ApiRef;
+  hasError: boolean;
+  componentProps?: any[];
+}
+
+export class ErrorBoundary extends React.Component<ErrorBoundaryProps, {}> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    if (this.props.api.current) {
```

Comment:
It feels like we add a lot of checks against `this.props.api.current` to make TypeScript happy but that unnecessary if not potentially hiding issues later down the road.